src/middleware/mongoQuery/index.js

```diff
@@ -0,0 +1,138 @@
+import _ from 'lodash';
+import * as mongodb from 'mongodb';
+import BadRequestError from '../../Errors/BadRequestError';
+
+/**
+ * Ensure that passed values are of the type indicated.
+ */
+var casters = {
+  'String': function(val) {
+    return `${val}`;
+  },
+  'ObjectId': function(val) {
+    if (mongodb.ObjectId.isValid(val)) {
+      return mongodb.ObjectId(val);
```

Comment:
According to http://mongodb.github.io/node-mongodb-native/2.1/api/ObjectID.html you should use `new` to create a new ObjectId

Reply:
:+1: